.changeset/lucky-clocks-love.md

```diff
@@ -0,0 +1,5 @@
+---
+"@latticexyz/cli": major
+---
+
+Removes `.mudbackup` file handling and `--backup`, `--restore, `--force` options from `mud set-version` command. Git will do a better job of this!
```

Comment:
Should we add some quick instructions for how to do the same with `git` to the changelog?

e.g.

```sh
# restore all package.json to the version on <commit>
git checkout <commit> -- package.json && \
git checkout <commit> packages/contracts/package.json && \
git checkout <commit> packages/client/package.json

# restore all package.json the version on the `main` branch
git checkout main -- package.json && \
git checkout main packages/contracts/package.json && \
git checkout main packages/client/package.json
```

(Note i haven't actually tested these, we should make sure this works)

Now that i'm typing this out i wonder if git is actually doing a better job here? What if more packages changed in the meantime? (Is there another way to do this via git that i'm missing here?)

Reply:
Ultimately I don't really think it's our job to figure out a user's upgrade path and how they resolve diffs in their files. In my mind, `set-version` is and should be just a shortcut for doing something like

```
pnpm up -r "@latticexyz/*@next"
```

except it takes into account all packages (not all of ours are prefixed like this) and has support for linking all packages too.